python-avd/pyavd/_eos_designs/structured_config/network_services/ip_igmp_snooping.py

```diff
@@ -87,41 +85,32 @@ def _ip_igmp_snooping_vlan(self: AvdStructuredConfigNetworkServices, vlan: dict,
             igmp_snooping_querier_enabled = True
 
         else:
-            igmp_snooping_enabled = vlan.get("igmp_snooping_enabled")
+            igmp_snooping_enabled = vlan.igmp_snooping_enabled
             if self.shared_utils.network_services_l3 and self.shared_utils.uplink_type in ["p2p", "p2p-vrfs"]:
-                igmp_snooping_querier_enabled = default(
-                    igmp_snooping_querier.get("enabled"),
-                    tenant_igmp_snooping_querier.get("enabled"),
-                )
+                igmp_snooping_querier_enabled = default(vlan.igmp_snooping_querier.enabled, tenant.igmp_snooping_querier.enabled)
 
         ip_igmp_snooping_vlan = {}
         if igmp_snooping_enabled is not None:
             ip_igmp_snooping_vlan["enabled"] = igmp_snooping_enabled
 
         if igmp_snooping_querier_enabled is not None:
             ip_igmp_snooping_vlan["querier"] = {"enabled": igmp_snooping_querier_enabled}
-            if igmp_snooping_querier_enabled is True:
-                address = default(igmp_snooping_querier.get("source_address"), tenant_igmp_snooping_querier.get("source_address"), self.shared_utils.router_id)
+            if igmp_snooping_querier_enabled:
+                address = default(vlan.igmp_snooping_querier.source_address, tenant.igmp_snooping_querier.source_address, self.shared_utils.router_id)
                 if address is not None:
                     ip_igmp_snooping_vlan["querier"]["address"] = address
 
-                version = default(
-                    igmp_snooping_querier.get("version"),
-                    tenant_igmp_snooping_querier.get("version"),
-                )
+                version = default(vlan.igmp_snooping_querier.version, tenant.igmp_snooping_querier.version)
                 if version is not None:
                     ip_igmp_snooping_vlan["querier"]["version"] = version
```

Comment:
it could be simplified using `remove_empty_from_dict`

Reply:
done